toolchain/lower/file_context.cpp

```diff
@@ -324,41 +329,59 @@ auto FileContext::BuildFunctionDefinition(SemIR::FunctionId function_id)
       sem_ir().inst_blocks().GetOrEmpty(function.implicit_param_refs_id);
   auto param_refs = sem_ir().inst_blocks().GetOrEmpty(function.param_refs_id);
   int param_index = 0;
-  if (SemIR::ReturnTypeInfo::ForFunction(sem_ir(), function, specific_id)
-          .has_return_slot()) {
-    function_lowering.SetLocal(function.return_slot_id,
-                               llvm_function->getArg(param_index));
-    ++param_index;
-  }
+  // The SemIR calling-convention parameters of the function, in order of
+  // runtime index. This is a transitional step toward generating this list
+  // in the check phase, which is why we're using the runtime index order
+  // even though it's less convenient for this usage.
```

Comment:
You don't seem to really be making use of the fact that this is in SemIR calling convention order. If you instead built a list in LLVM IR calling convention order, I think the `lower_param` code could instead be a loop over that array. Perhaps you could factor out the code in `BuildFunctionDecl` that constructs `param_inst_ids`, since that's then the list that we'd want here.

Reply:
The next PR is going to build `calling_convention_param_ids` in `CalleePatternMatch`, and propagate that through instead of `param_refs` and `implicit_param_refs`. At that point it would be difficult and unnatural to arrange for `calling_convention_param_ids` to be in LLVM order. Consequently, I'm intentionally creating it in SemIR order now, in order to validate the code below that manages the transition from SemIR order to LLVM order.

Reply:
OK, thanks for explaining, that makes sense :)